Doc/c-api/intro.rst

```diff
@@ -838,3 +838,44 @@ after every statement run by the interpreter.)
 
 Please refer to :file:`Misc/SpecialBuilds.txt` in the Python source distribution
 for more detailed information.
+
+
+.. _c-api-tools:
+
+Recommended third party tools
+=============================
+
+The following third party tools offer both simpler and more sophisticated
+approaches to creating C and C++ extensions for Python:
+
+* `Cython <https://cython.org/>`_
+* `cffi <https://cffi.readthedocs.io>`_
+* `HPy <https://hpyproject.org/>`_
+* `nanobind <https://github.com/wjakob/nanobind>`_ (C++)
+* `Numba <https://numba.pydata.org/>`_
+* `pybind11 <https://pybind11.readthedocs.io/>`_ (C++)
+* `PyO3 <https://pyo3.rs/>`_ (Rust)
+* `SWIG <https://www.swig.org>`_
+
+Using tools such as these can help avoid writing code that is tightly bound to
+a particular version of CPython, avoid reference counting errors, and focus
+more on your own code than on using the CPython API. In general, new versions
+of Python can be supported by updating the tool, and your code will often use
+newer and more efficient APIs automatically. Some tools also support compiling
+for other implementations of Python from a single set of sources.
+
+Directly coding against the CPython C API should, in most cases, be your last
+resort rather than the first.
```

Comment:
+1 without this paragraph. 

There are good reasons against using the C API directly, and they're spelled out above. But there are also reasons *for* using it -- notably, not bringing in another dependency with its another maintenance/support timeline/mechanism.

```suggestion
```

I'd be happier if we didn't maintain a list of third-party projects in the docs, but since we already have (an outdated) one, this PR is an improvement.